csi/server/plugin/opensds/controller.go

```diff
@@ -498,11 +498,21 @@ func (p *Plugin) CreateSnapshot(
 		return nil, status.Error(codes.InvalidArgument, "Source Volume ID cannot be empty")
 	}
 
+
+
 	snapReq := &model.VolumeSnapshotSpec{
 		Name:     req.Name,
 		VolumeId: req.SourceVolumeId,
 	}
 
+	for k, v := range req.GetParameters() {
+		switch strings.ToLower(k) {
+		case KParamProfile:
+			snapReq.ProfileId = v
```

Comment:
Profile name should be supported too.  
At least add a TODO here to support profile name.